web-bundle/src/main/java/com/graphhopper/http/TruckFlagEncoder.java

```diff
@@ -0,0 +1,31 @@
+package com.graphhopper.http;
+
+import com.graphhopper.routing.ev.*;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.VehicleEncodedValues;
+
+// adapted from prod GraphHopper code (not available in OSS GraphHopper)
+public class TruckFlagEncoder {
+    public static final String TRUCK_VEHICLE_NAME = "truck";
+    private static final int TRUCK_SPEED_BITS = 6;
+    private static final int TRUCK_SPEED_FACTOR = 2;
+    private static final boolean ENABLE_TRUCK_TURN_RESTRICTIONS = false;
+
+    public static FlagEncoder createTruckFlagEncoder() {
+        // turn costs is binary -- restricted or unrestricted (1 is scaled to infinity further down the code)
```

Comment:
these comments were adapted from Michael's in https://github.com/replicahq/graphhopper/commit/fb3c251196d6fb971c54b8628eecb7b107c1b59d and https://github.com/replicahq/graphhopper/commit/96497f0a42611eb8c60dca0c8eda011a784514f0 (same with the comment about configuration in ReplicaFlagEncoderFactory)